pkg/util/validation/limits.go

```diff
@@ -208,8 +209,9 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxChunksPerQuery, MaxChunksPerQueryFlag, 2e6, "Maximum number of chunks that can be fetched in a single query from ingesters and long-term storage. This limit is enforced in the querier, ruler and store-gateway. 0 to disable.")
 	f.IntVar(&l.MaxFetchedSeriesPerQuery, MaxSeriesPerQueryFlag, 0, "The maximum number of unique series for which a query can fetch samples from each ingesters and storage. This limit is enforced in the querier and ruler. 0 to disable")
 	f.IntVar(&l.MaxFetchedChunkBytesPerQuery, MaxChunkBytesPerQueryFlag, 0, "The maximum size of all chunks in bytes that a query can fetch from each ingester and storage. This limit is enforced in the querier and ruler. 0 to disable.")
-	f.Var(&l.MaxQueryLength, maxQueryLengthFlag, "Limit the query time range (end - start time). This limit is enforced in the querier (on the query possibly split by the query-frontend) and ruler. 0 to disable.")
-	f.Var(&l.MaxPartialQueryLength, "querier.max-partial-query-length", fmt.Sprintf("Limit the time range for partial queries at the querier level. Defaults to the value of -%s if set to 0.", maxQueryLengthFlag))
+	// TODO: Deprecated in Mimir 2.6, remove in Mimir 2.8
+	f.Var(&l.MaxQueryLength, maxQueryLengthFlag, fmt.Sprintf("Deprecated: Limit the query time range (end - start time). This limit is enforced in the querier (on the query possibly split by the query-frontend) and ruler. 0 to disable. This option is deprecated, use -%s or -%s instead.", maxPartialQueryLengthFlag, maxTotalQueryLengthFlag))
```

Comment:
We normally use `flagext.DeprecatedFlag` to deprecate flags. This also increases `deprecated_flags_inuse_total` counter, which users can use to check if they use deprecated options.

Reply:
Unfortunatley `DeprecatedFlag` doesn't set the actual value on the config. Here we need to keep backward compatibility.

Reply:
Right, I was going to use `DeprecatedFlag` but we're still using `l.MaxQueryLength` as a fallback for the new options (max partial length and max total length).